gene2phenotype_project/gene2phenotype_project/settings.py

```diff
@@ -79,23 +85,55 @@
 REST_FRAMEWORK = {
     'DEFAULT_PAGINATION_CLASS': 'rest_framework.pagination.LimitOffsetPagination',
     'PAGE_SIZE': 50,
-    'DEFAULT_AUTHENTICATION_CLASSES': ('knox.auth.TokenAuthentication',),
+    'DEFAULT_AUTHENTICATION_CLASSES': ('rest_framework_simplejwt.authentication.JWTAuthentication', 'gene2phenotype_app.authentication.CustomAuthentication'),
     'DEFAULT_PERMISSION_CLASSES': (
         "rest_framework.permissions.AllowAny",
     ),
 }
 
-
-REST_KNOX = {
+SIMPLE_JWT = {
   'SECURE_HASH_ALGORITHM': 'hashlib.sha512',
   'AUTH_TOKEN_CHARACTER_LENGTH': 64,
-  'TOKEN_TTL': timedelta(hours=8), # time before it logs you out 
-  'USER_SERIALIZER': 'gene2phenotype_app.serializers.UserSerializer',
+  'TOKEN_USER_CLASS': 'gene2phenotype_app.serializers.UserSerializer',
   'TOKEN_LIMIT_PER_USER': None,
+  "USER_ID_FIELD": "id",
+  "USER_ID_CLAIM": "id",
   'AUTO_REFRESH': False,
   'EXPIRY_DATETIME_FORMAT': api_settings.DATETIME_FORMAT,
+  ##Custom
+  "AUTH_COOKIE": "access_token",
+  "REFRESH_COOKIE": "refresh_token",
+  "AUTH_COOKIE_DOMAIN": None,
+  "AUTH_COOKIE_SECURE": False, # for production, we need to change this to true
+  "AUTH_COOKIE_HTTP_ONLY": True, #prevents client side js from accessing the cookie
+  "AUTH_COOKIE_PATH": "/",
+  "AUTH_COOKIE_SAMESITE": "Lax",
+  "ACCESS_TOKEN_LIFETIME": timedelta(hours=1),
+  "REFRESH_TOKEN_LIFETIME": timedelta(days=1),
+  "BLACKLIST_AFTER_ROTATION": True,
+  'TOKEN_BACKEND': 'rest_framework_simplejwt.token_blacklist.backends.BlacklistBackend',
+  'BLACKLIST_TOKEN_CHECKS': [ 'rest_framework_simplejwt.token_blacklist.check_blacklisted_token', ],
+  "ROTATE_REFRESH_TOKENS": True
 }
 
+CORS_ALLOWED_ORIGINS = [
+    "http://localhost:5173",  # Add your frontend's URL here, # local set up still being implemented, need to change for production
+]
+CSRF_TRUSTED_ORIGINS = [
+    'http://localhost:5173',  # Add your frontend's URL here, # local set up still being implemented, need to change for production
+]
+CORS_ALLOWED_CREDENTIALS = "True"
```

Comment:
Before the PR is merged, this could be removed.
We'll update the variables in the VM.

Reply:
Yeah, i intend to remove it, kept it there for local testing